### PR TITLE
Add some Guile packages

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
##### Motivation for this change
They're needed for building GNU Guix from source ;)

Guix 1.0 already builds with this, but needs a little more tweaking.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/guile-modules/guile-ssh/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "artyom-poptsov";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "054hd9rzfhb48gc1hw3rphhp0cnnd4bs5qmidy5ygsyvy9ravlad";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig which texinfo ];  # Which is probably a hack
+  buildInputs = [ guile libssh ]; # Maybe use libssh2
```
Comment (doronbehar) [inline pkgs/development/guile-modules/guile-ssh/default.nix]: > Maybe use libssh2

Just out of curiosity, what did you mean?

File: pkgs/development/guile-modules/guile-bytestructures/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, autoreconfHook, pkgconfig, guile }:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-bytestructures/default.nix:2]:
```suggestion

stdenv.mkDerivation rec {
```

File: pkgs/development/guile-modules/guile-bytestructures/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  version = "1.0.7";
+
+  src = fetchFromGitHub {
+    owner = "TaylanUB";
+    repo = "scheme-bytestructures";
+    rev = "v${version}";
+    sha256 = "0q0habjiy3h9cigb7q1br9kz6z212dn2ab31f6dgd3rrmsfn5rvb";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-bytestructures/default.nix:13]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config ];
```

File: pkgs/development/guile-modules/guile-bytestructures/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, autoreconfHook, pkgconfig, guile }:
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-bytestructures/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, autoreconfHook, pkg-config, guile }:
```

File: pkgs/development/guile-modules/guile-gcrypt/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+stdenv.mkDerivation rec {
+  pname = "guile-gcrypt";
+  version = "0.2.1";
+
+  src = fetchurl {
+    url = "https://notabug.org/cwebber/${pname}/archive/v${version}.tar.gz";
+    sha256 = "1qj1yw0kman984x584jjjxnjdhm0pwgp09iyn3b5rqajx7svpqcd";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig texinfo ];
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-gcrypt/default.nix:11]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config texinfo ];
```

File: pkgs/development/guile-modules/guile-gcrypt/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchurl, autoreconfHook, pkgconfig, texinfo, guile, libgcrypt }:
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-gcrypt/default.nix:1]:
```suggestion
{ stdenv, fetchurl, autoreconfHook, pkg-config, texinfo, guile, libgcrypt }:

```

File: pkgs/development/guile-modules/guile-git/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitLab {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1s77s70gzfj6h7bglq431kw8l4iknhsfpc0mnvcp4lkhwdcgyn1n";
+  };
+
+  GUILE_AUTO_COMPILE = 0;
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig texinfo ];
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-git/default.nix:17]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config texinfo ];
```

File: pkgs/development/guile-modules/guile-git/default.nix [hunk 8]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitLab, autoreconfHook, pkgconfig, texinfo
+, guile, guile-bytestructures, libgit2
+}:
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-git/default.nix:3]:
```suggestion
}:

```

File: pkgs/development/guile-modules/guile-git/default.nix [hunk 9]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitLab, autoreconfHook, pkgconfig, texinfo
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-git/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitLab, autoreconfHook, pkg.config, texinfo
```

File: pkgs/development/guile-modules/guile-json/default.nix [hunk 10]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchurl, autoreconfHook, pkgconfig, guile }:
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-json/default.nix:1]:
```suggestion
{ stdenv, fetchurl, autoreconfHook, pkg-config, guile }:

```

File: pkgs/development/guile-modules/guile-json/default.nix [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+stdenv.mkDerivation rec {
+  pname = "guile-json";
+  version = "3.3.0";
+
+  src = fetchurl {
+    url = "mirror://savannah/${pname}/${pname}-${version}.tar.gz";
+    sha256 = "0gbqiii8yl5vg5d2gf8j6pd1wwy9c37sxpdvv94rqnnp11rkbdyf";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-json/default.nix:11]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config ];
```

File: pkgs/development/guile-modules/guile-sqlite3/default.nix [hunk 12]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchurl, autoreconfHook, pkgconfig, guile, sqlite }:
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-sqlite3/default.nix:1]:
```suggestion
{ stdenv, fetchurl, autoreconfHook, pkg-config, guile, sqlite }:

```

File: pkgs/development/guile-modules/guile-sqlite3/default.nix [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+stdenv.mkDerivation rec {
+  pname = "guile-sqlite3";
+  version = "0.1.0";
+
+  src = fetchurl {
+    url = "https://notabug.org/${pname}/${pname}/archive/v${version}.tar.gz";
+    sha256 = "1s9gmj72vszn999c82vnq0vic7ly4wcg8lz1qcfmhgk9pihcs0bm";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-sqlite3/default.nix:11]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config ];
```

File: pkgs/development/guile-modules/guile-ssh/default.nix [hunk 14]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, fetchFromGitHub, fetchpatch, autoreconfHook, pkgconfig, which, texinfo, guile, libssh }:
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-ssh/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, fetchpatch, autoreconfHook, pkg-config, which, texinfo, guile, libssh }:

```

File: pkgs/development/guile-modules/guile-ssh/default.nix [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+  version = "0.12.0";
+
+  src = fetchFromGitHub {
+    owner = "artyom-poptsov";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "054hd9rzfhb48gc1hw3rphhp0cnnd4bs5qmidy5ygsyvy9ravlad";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig which texinfo ];  # Which is probably a hack
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-ssh/default.nix:13]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config which texinfo ];
```

File: pkgs/development/guile-modules/guile-ssh/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-ssh/default.nix:14]: On what depends this decision?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please use fetchgit for getting archives from https://notabug.org.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 79650` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>guile-bytestructures</li>
    <li>guile-gcrypt</li>
    <li>guile-git</li>
    <li>guile-json</li>
    <li>guile-sqlite3</li>
    <li>guile-ssh</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 79650` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>guile-bytestructures</li>
    <li>guile-gcrypt</li>
    <li>guile-git</li>
    <li>guile-json</li>
    <li>guile-sqlite3</li>
    <li>guile-ssh</li>
  </ul>
</details>

**Comment by jgarte** [comment]
Hi, was the only think blocking this pull request the notabug change?

I'd be happy to work on this in order to finish adding these packages.

**Comment by SuperSandro2000** [comment]
> Hi, was the only think blocking this pull request the notabug requested url change?
> 
> I'd be happy to work on this in order to finish adding these packages.

no, there a few other things which now no longer work.

**Comment by doronbehar** [comment]
Closing due to not response from OP.

**Pull request closed without merging**